File: src/mundo/config/arvores.js
// O `id` continua sendo o mesmo slug da Deidade (usado nas rotas /deidades/{id},
// no localStorage e nos nomes dos .glb) — só o `titulo` (nome da Árvore, exibido
// no globo 3D) e o `rgb` (cor da Árvore) mudaram, seguindo a tabela de
// referência "Estrutura do Jardim". A Deidade (Aethel, Ousias...) continua
// sendo o nome usado nas páginas de detalhe/prosa importadas.
//
// `keryx` é um caso especial: a Árvore Parley foi subjugada pela A.X.I.S (o
// Fluxo tecnológico de Jota Macedo) — fisicamente é a mesma Árvore/local, só
// que agora dominada por uma gaiola artificial (ver modeloId: 'axis' e
// tools/blender/generate_tree.py --mode axis). `titulo`/`rgb` passam a
// refletir a identidade dominante (A.X.I.S, neon) em todo canto que já lê
// esses campos (breadcrumbs, cor de destaque nas páginas de detalhe...);
// `tituloSubjugada`/`rgbSubjugada` guardam a identidade original (Parley,
// prata) só pra o efeito de reflexo/glitch no rótulo 3D (ver tree3d.js) —
// um sinal de que Parley ainda existe, fraca, por baixo do controle da A.X.I.S.
//
// `mulher-carmesim` também é um caso especial: `isolada: true` faz a cena 3D
// (tree3d.js) plantar a cúpula dela num canto fixo e isolado, fora do
// espalhamento/relaxamento comum das outras Árvores — reflete a lore (ela "é
// e não é uma Árvore", representa o Fim de tudo).
export const ARVORES = [
  { id: 'aethel',          titulo: 'Gênese',   rgb: '214,120,156' }, // Rosa
  { id: 'ousias',          titulo: 'Alétheia', rgb: '222,198,88'  }, // Amarelo
  {
    id: 'keryx', titulo: 'A.X.I.S', rgb: '53,216,236', // Azul-neon artificial
    modeloId: 'axis', subjugada: true,
    tituloSubjugada: 'Parley', rgbSubjugada: '192,198,206', // Prata (identidade original)
  },
  { id: 'haemus',          titulo: 'Anima',    rgb: '86,172,92'   }, // Verde
  { id: 'ignis',           titulo: 'Vórtice',  rgb: '222,114,42'  }, // Laranja
  { id: 'moros',           titulo: 'Baluarte', rgb: '116,82,52'   }, // Marrom
  { id: 'aperion',         titulo: 'Matriz',   rgb: '132,84,188'  }, // Roxo
  { id: 'chronus',         titulo: 'Éon',      rgb: '168,138,72'  }, // Dourado envelhecido
  { id: 'erebus',          titulo: 'Abismo',   rgb: '34,30,40'    }, // Preto
  { id: 'mulher-carmesim', titulo: 'Limiar',   rgb: '134,28,48', isolada: true }, // Vermelho vinho
];

export function corDaArvore(arvoreId) {
  return ARVORES.find(arvore => arvore.id === arvoreId)?.rgb || '160,160,170';
}

File: bots/jornalista/core/arvores.py
"""
As 10 Árvores do Jardim — nome e cor oficiais, mesma fonte que
src/mundo/config/arvores.js usa pra cena 3D (RGB convertido pra hex aqui).

Usado pelo registro por Árvore (Passo 5 do Plano_Jornalista.md, Decisão 1):
puramente cosmético, "qual Árvore você é" — sem ligação com classe/ficha.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import List, Optional


@dataclass(frozen=True)
class Arvore:
    id: str
    nome: str
    cor: int  # hex, pronto pra discord.Colour(...)


ARVORES: List[Arvore] = [
    Arvore("aethel", "Gênese", 0xD6789C),
    Arvore("ousias", "Alétheia", 0xDEC658),
    Arvore("keryx", "A.X.I.S", 0x35D8EC),
    Arvore("haemus", "Anima", 0x56AC5C),
    Arvore("ignis", "Vórtice", 0xDE722A),
    Arvore("moros", "Baluarte", 0x745234),
    Arvore("aperion", "Matriz", 0x8454BC),
    Arvore("chronus", "Éon", 0xA88A48),
    Arvore("erebus", "Abismo", 0x221E28),
    Arvore("mulher-carmesim", "Limiar", 0x861C30),
]

_POR_ID = {a.id: a for a in ARVORES}


def obter(arvore_id: str) -> Optional[Arvore]:
    return _POR_ID.get(arvore_id)
